src/python/lib/clusterfuzz/stacktraces/__init__.py

```diff
@@ -0,0 +1,1362 @@
+      # anonymous namespaces in the crash state.
+      if (frame_struct and frame_struct.module_offset and
+          not frame_struct.function_name):
+        frame_filter = lambda s: s
+
+      # Update stacktrace frames list with frame struct.
+      new_thread = state.last_frame_id < 0
+      try:
+        # We have a 'frame_id' group, so pull the ID and check if we need to
+        # start a new thread. Also update last_frame_id accordingly.
+        frame_id = int(match.group('frame_id').strip())
+        if frame_id < state.last_frame_id:
+          new_thread = True
+        state.last_frame_id = frame_id
+      except IndexError:
+        # If there is not 'frame_id' group, just stick everything into one
+        # thread.
+        state.last_frame_id += 1
+
+      if new_thread:
+        state.frames.append([])
+      state.frames[-1].append(frame_struct)
+
+    if frame_override_func:
+      frame = frame_override_func(frame, frame_struct)
+
+    # If we are ignoring a frame, we still have a match. Don't add it to the
+    # state, but notify the caller that we found something.
+    if can_ignore and self.ignore_stack_frame(frame):
+      return match
+
+    # Filter the frame and add to a list.
+    filtered_frame = frame_filter(frame)
+    state.raw_frames.append(filtered_frame)
+
+    # Update the crash state only if we need more frames.
+    if state.frame_count < MAX_CRASH_STATE_FRAMES:
+      state.crash_state += filtered_frame + '\n'
+      state.frame_count += 1
+
+    return match
+
+  def update_state_on_check_failure(self, state, line, regex, crash_type):
+    """Update the state if the crash is a CHECK failure."""
+    check_match = self.update_state_on_match(
+        regex, line, state, new_type=crash_type, reset=True, new_frame_count=1)
+    if check_match:
+      failure_string = fix_check_failure_string(check_match.group(2))
+      source_file = fix_filename_string(check_match.group(1))
+      state.crash_state = '%s in %s\n' % (failure_string, source_file)
+
+  def match_assert(self, line, state, regex, group=1):
+    """Match an assert."""
+    assert_match = self.update_state_on_match(
+        regex, line, state, new_type='ASSERT', new_frame_count=1)
+    if assert_match and assert_match.group(group):
+      # For asserts, we want to actually use the match as the crash state.
+      state.crash_state = assert_match.group(group) + '\n'
+
+  def filter_crash_parameters(self, state):
+    """Normalize crash parameters into generic format regardless of the tool
+    used."""
+    # Filter crash state represented in |state|.
+    # Remove non-printable chars from crash state.
+    state.crash_state = ''.join(
+        s for s in state.crash_state if s in string.printable)
+
+    # Shorten JNI messages.
+    if JNI_ERROR_STRING in state.crash_state:
+      state.crash_state = state.crash_state.replace(JNI_ERROR_STRING, 'JNI:')
+
+    if self.symbolized:
+      # 1. Normalize addresses and numbers in crash_state.
```

Comment:
nit: maybe lets remove these 1., 2., since missing at end, can remove.

Reply:
Removed the numbers!